Modules/Filtering/ImageFilterBase/test/itkCastImageFilterTest.cxx

```diff
@@ -400,6 +400,52 @@ TestVectorImageCast2()
 }
 
 
+bool
+TestVectorImageCast3()
+{
+  // This function casts a VectorImage<float, 2>
+  // to an Image<Vector<unsigned char, 2>, 2>
+  std::cout << "Casting from an Image<Vector<float,3>, 3> to VectorImage<float, 3> ... ";
+
+  using VectorFloatImageType = itk::Image<itk::Vector<float, 3>, 3>;
+  using FloatVectorImageType = itk::VectorImage<float, 3>;
+
+  // Create a 1x3 image of 2D vectors
+  auto image = VectorFloatImageType::New();
+
+  constexpr itk::Size<3>    size{ { 1, 1, 4 } };
+  const itk::ImageRegion<3> region{ size };
+  image->SetRegions(region);
+  image->AllocateInitialized();
+  VectorFloatImageType::PixelType vec;
+
+  // Only the first pixel will be the vector (1.3, 5.3, 6.4)
+  vec[0] = 1.3;
+  vec[1] = 5.3;
+  vec[2] = 6.4;
+  image->SetPixel(itk::Index<3>(), vec);
+
+  using CastImageFilterType = itk::CastImageFilter<VectorFloatImageType, FloatVectorImageType>;
+  auto castImageFilter = CastImageFilterType::New();
+  castImageFilter->SetInput(image);
+  castImageFilter->SetNumberOfWorkUnits(1);
+  castImageFilter->Update();
+
+  bool success = true;
+  if (std::memcmp(castImageFilter->GetOutput()->GetBufferPointer(), image->GetBufferPointer(), 4) == 0)
```

Comment:
Why are only the first `4` bytes compared? I guess it's because it is `sizeof(float)`, right? If so, please replace `4` with  `sizeof(float)`.